spring-cloud-kubernetes-integration-tests/spring-cloud-kubernetes-client-config-it/src/test/java/org/springframework/cloud/kubernetes/client/config/it/ConfigMapAndSecretIT.java

```diff
@@ -78,17 +74,35 @@
 
 	private static K8SUtils k8SUtils;
 
+	private static final K3sContainer K3S = new K3sContainer(DockerImageName.parse("rancher/k3s:v1.21.10-k3s1"))
+			.withFileSystemBind("/tmp/images", "/tmp/images", BindMode.READ_WRITE).withExposedPorts(80, 6443)
+			.withCommand("server") // otherwise, traefik is not installed
+			.withReuse(true);
+
 	@BeforeAll
-	public static void setup() throws Exception {
-		client = createApiClient();
+	static void setup() throws Exception {
+		K3S.start();
+		K3S.execInContainer("ctr", "i", "import", "/tmp/images/spring-cloud-kubernetes-client-config-it.tar");
+		createApiClient(K3S.getKubeConfigYaml());
 		api = new CoreV1Api();
 		appsApi = new AppsV1Api();
 		networkingApi = new NetworkingV1Api();
 		k8SUtils = new K8SUtils(api, appsApi);
+
+		RbacAuthorizationV1Api rbacApi = new RbacAuthorizationV1Api();
+		api.createNamespacedServiceAccount(NAMESPACE, getConfigK8sClientItServiceAccount(), null, null, null);
+		rbacApi.createNamespacedRoleBinding(NAMESPACE, getConfigK8sClientItRoleBinding(), null, null, null);
+		rbacApi.createNamespacedRole(NAMESPACE, getConfigK8sClientItRole(), null, null, null);
+	}
+
+	@AfterAll
+	static void afterAll() throws Exception {
```

Comment:
unload the image, once we are done. this is needed because we enable `withReuse` and tests might step on each other.